sklearn/base.py

```diff
@@ -392,6 +393,52 @@ def _check_n_features(self, X, reset):
                 f"is expecting {self.n_features_in_} features as input."
             )
 
+    def _check_feature_names(self, X, reset=True):
```

Comment:
It's a bit weird to me to have this be one function with completely different behavior depending on ``reset`` (there is no code shared between the two options) but I guess it makes sense if it's being called from ``_validate_data``...